contracts/BC_fusion/StakePool.sol

```diff
@@ -0,0 +1,290 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/CountersUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/structs/DoubleEndedQueueUpgradeable.sol";
+
+import "./System.sol";
+
+interface IStakeHub {
+    function unbondPeriod() external view returns (uint256);
+    function transferGasLimit() external view returns (uint256);
+    function poolImplementation() external view returns (address);
```

Comment:
No need to have `poolImplementation` here.

Reply:
Removed